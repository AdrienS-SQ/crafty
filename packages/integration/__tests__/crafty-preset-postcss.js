/* global describe, it, expect */

const fs = require("fs");
const path = require("path");

const rimraf = require("rimraf");
const configuration = require("@swissquote/crafty/src/configuration");
const getCommands = require("@swissquote/crafty/src/commands/index");

const testUtils = require("../utils");

const getCrafty = configuration.getCrafty;

it("Loads crafty-preset-postcss and does not register gulp tasks", () => {
  const crafty = getCrafty(["@swissquote/crafty-preset-postcss"], {});

  const loadedPresets = crafty.config.loadedPresets.map(
    preset => preset.presetName
  );
  expect(loadedPresets).toContain("@swissquote/crafty-preset-postcss");

  const commands = getCommands(crafty);
  expect(Object.keys(commands)).toContain("cssLint");

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([]);
});

it("Lints with the command", async () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-postcss/no-bundle")
  );
  rimraf.sync("dist");

  const result = await testUtils.run(["cssLint", "css/*.scss"]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist")).toBeFalsy();
});

it("Lints with the command in legacy mode", async () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-postcss/no-bundle")
  );
  rimraf.sync("dist");

  const result = await testUtils.run(["cssLint", "css/*.scss", "--preset", "legacy"]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist")).toBeFalsy();
});

it("Lints with the command with custom config", async () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-postcss/no-bundle")
  );
  rimraf.sync("dist");

  const result = await testUtils.run([
    "cssLint",
    "css/*.scss",
    "--config",
    "stylelint.json"
  ]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist")).toBeFalsy();
});

it("Creates IDE Integration files", async () => {
  process.chdir(path.join(__dirname, "../fixtures/crafty-preset-postcss/ide"));
  rimraf.sync("stylelint.config.js");
  rimraf.sync("prettier.config.js");
  rimraf.sync(".gitignore");

  const result = await testUtils.run(["ide"]);

  expect(result).toMatchSnapshot();
  expect(
    testUtils.snapshotizeOutput(
      fs.readFileSync("stylelint.config.js").toString("utf8")
    )
  ).toMatchSnapshot();

  expect(
    testUtils.snapshotizeOutput(
      fs.readFileSync("prettier.config.js").toString("utf8")
    )
  ).toMatchSnapshot();

  expect(
    testUtils.snapshotizeOutput(fs.readFileSync(".gitignore").toString("utf8"))
  ).toMatchSnapshot();
});
