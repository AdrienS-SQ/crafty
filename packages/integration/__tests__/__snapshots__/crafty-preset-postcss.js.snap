// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lints with the command 1`] = `
Object {
  "status": 2,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...

css/imported.scss
 1:1  ✖  Types are allowed only inside a scope  swissquote/no-type-outside-scope

css/style.scss
  4:1   ✖  Types are allowed only inside a scope                                 swissquote/no-type-outside-scope
  6:14  ✖  Unexpected unit                                                       length-zero-no-unit
 10:1   ✖  Expected class selector \\".notAComponent\\" to match specified pattern   selector-class-pattern
 15:1   ✖  A state must be linked to a component                                 swissquote/no-state-without-component
 20:1   ✖  Expected \\"#no-id\\" to have no more than 0 id selectors                 selector-max-id
 28:13  ✖  Expected nesting depth to be no more than 2                           max-nesting-depth
 31:5   ✖  Unexpected missing end-of-source newline                              no-missing-end-of-source-newline

",
}
`;

exports[`Lints with the command in legacy mode 1`] = `
Object {
  "status": 2,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...

css/style.scss
  6:14  ✖  Unexpected unit                            length-zero-no-unit
 31:5   ✖  Unexpected missing end-of-source newline   no-missing-end-of-source-newline

",
}
`;

exports[`Lints with the command with custom config 1`] = `
Object {
  "status": 2,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...

css/style.scss
  6:14  ✖  Unexpected unit                                                       length-zero-no-unit
 10:1   ✖  Expected class selector \\".notAComponent\\" to match specified pattern   selector-class-pattern
 15:1   ✖  A state must be linked to a component                                 swissquote/no-state-without-component
 20:1   ✖  Expected \\"#no-id\\" to have no more than 0 id selectors                 selector-max-id
 28:13  ✖  Expected nesting depth to be no more than 2                           max-nesting-depth
 31:5   ✖  Unexpected missing end-of-source newline                              no-missing-end-of-source-newline

",
}
`;
